File: Lourd/PolyPaint/PolyPaint/Templates/LinkStyle.cs
namespace PolyPaint.Templates
{
    public class LinkStyle
    {
        public string color { get; set; }
        public int thickness { get; set; }
        public int type { get; set; }

        public LinkStyle()
        {
        }

        public LinkStyle(string color, int thickness, int type)
        {
            this.color = color;
            this.thickness = thickness;
            this.type = type;
        }

        public LinkStyle(LinkStyle style)
        {
            color = style.color;
            thickness = style.thickness;
            type = style.type;
        }

        public void SetDefaults()
        {
            color = "#FF000000";
            thickness = 2;
            type = 0; //(int)StrokeTypes.LINK;
        }
    }
}

File: Leger/PolyPaint/app/src/main/java/com/polypaint/polypaint/Model/LinkStyle.kt
package com.polypaint.polypaint.Model
import java.io.Serializable

class LinkStyle(var color: String, var thickness: Int, var type: Int): Serializable{

    fun copy(color: String=this.color,  thickness: Int=this.thickness,  type: Int=this.type) = LinkStyle(color, thickness, type)
}